compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1715,6 +1715,8 @@ object desugar {
       case PatDef(mods, pats, tpt, rhs) =>
         val pats1 = if (tpt.isEmpty) pats else pats map (Typed(_, tpt))
         flatTree(pats1 map (makePatDef(tree, mods, _, rhs)))
+      case ext: ExtMethods =>
+        Block(List(ext), Literal(Constant(())).withSpan(ext.span))
```

Comment:
If this can only happen in case of error, maybe the parser should be changed to not try to parse extension methods in such positions?

Reply:
Why would we special case extensions? Isn't it better to handle them the same as `def`s? I would expect that the following would befave similarly

```scala

def f1: Unit = { def g = 0 }
def f2: Unit = { extension (x: Int) def g = 0 }


```

Reply:
Also, same error should be emitted for

```scala
def f1: Int = { def g = 0 } // error: Found: Unit   Required: Int
def f2: Int = { extension (x: Int) def g = 0 } // error: Found: Unit   Required: Int
```

Reply:
Ah ok then is it possible to have a pos test for this change?

Reply:
Yes, ill add the test

Reply:
Added these tests